app/src/main/java/com/hover/stax/countries/CountryAdapter.java

```diff
@@ -0,0 +1,64 @@
+package com.hover.stax.countries;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatTextView;
+
+import com.hover.stax.R;
+import com.hover.stax.utils.UIHelper;
+
+public class CountryAdapter extends ArrayAdapter<String> {
+	private static final String TAG ="CountryAdapter";
+	private String[] countryCodes;
+
+	public CountryAdapter(@NonNull  String[] countryCodes, @NonNull Context context) {
+		super(context, 0, countryCodes);
+		this.countryCodes = countryCodes;
+
+	}
+
+	@SuppressLint("ViewHolder")
```

Comment:
If you instead just inflate the view only if it is null then it won't give you this ViewHolder error and you'll get the same performance as wit viewholder since it will reuse the views. You're probably better off not using a viewholder here anyway since it basically adds more boilerplate code. This is all fine as-is, this is just a note

Reply:
Thanks!